### ISSUE TITLE
when i use getter and setter, i got this, is this a issue? 

### ISSUE THREAD

**Issue by lornally** [stampit-org/stampit#344]
maybe this is issue, or maybe i use it by incorrect method, here is my code. 
```js
function getset() {
  const x = stampit()
    .props({ i: 'iiiii' })
    .methods({
      get info() {
        return `get${this.i}`;
      },
      set info(info) {
        this.i += `hahah${info}`;
      },
    });

  const t = x();
  t.info = 'ttttt';
  log('info:', t.info); //now  t.info is 'ttttt'  !!!!!! , it look like getter and setter is useless?
  log('i:', t.i);
}
```

**Comment by koresar** [comment]
Hello there!
Thank you for the issue. This is an unimplemented feature. It is requested
quite often. I guess it's time to implement it.
Sorry about that. I need to allocate some time to make it happen.

**Comment by lornally** [comment]
good, and maybe i can help you. :)

**Comment by koresar** [comment]
Superb! The code is very unreadable though.
The getters and setters should be copied to the ".methods" object while
composing.

**Comment by lornally** [comment]
yes, it is really tough, i am trying to understand it.......

**Comment by koresar** [comment]
Here is almost the same code but more readable.
https://github.com/stampit-org/stamp-specification/blob/master/compose.js

**Comment by lornally** [comment]
the code is really elegant, but i really can not understand why getter/setter is not work, the stampit copy all property of the object. but why getter/setter is not work? maybe i need to get more information about getter/setter.

**Comment by lornally** [comment]
my error, this code is ok, so the proto nest is not problem: 
```js
class X {
  constructor(name) {
    this._name = name;
  }
  get name() {
    return this._name;
  }
  set name(name) {
    this._name = `${name}X`;
  }
}
class Y extends X {}
const yy = new Y('');
yy.name = 'hi';
log('yyyyyyy', yy); //

```

**Comment by lornally** [comment]
maybe i have find the solution, because i find a big lack of my cognitive concept. 
- getter/setter is not method, it is properties.
because this doc : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty
so we can do like this: 
1. Object.getOwnPropertyDescriptors()  or Object.getOwnPropertyNames()
2. then Object​.define​Properties()

**Comment by koresar** [comment]
Sounds you're on the right track! Please, continue. 👍

**Comment by lornally** [comment]
yes, i have test 3way, code: 
```js
const y = {
  i: '-init-',
  get info() {
    return `read${this.i}`;
  },
  set info(info) {
    this.i += `write${info}`;
  },
};
log('0object0 info:', y.info);  //right
y.info = 'y';
log('1object1 info:', y.info);  //right too

//test create
let x=Object.create(y);
log('00x info:', x.info);  //pass
x.info='x'
log('11x info:', x.info); //pass
log('11x11 info:', x.pp());  //pass

//test assign
x=Object.assign(y);
log('22x info:', x.info);  //pass
x.info='x'
log('33x info:', x.info);  //pass
log('44x info:', x.pp());  //pass

//test spread...
x={...y};
log('55x info:', x.info);
x.info='x'
log('66x info:', x.info);  //here error.
log('77x info:', x.pp());
```
so   we can use assign or create to deal property, that can deal getter/setter in right way, now i will try to fix compose.js first.

**Comment by lornally** [comment]
this is the right code, maybe, i need change a mass of code, every property include deep and static:
```js
 function com(obj) {
    return function com() {
      const p = Object.defineProperties({}, Object.getOwnPropertyDescriptors(obj));
      return p;
    };
  }
```

**Comment by lornally** [comment]
this is demo:  com=compose
```js
export function com(obj) {
	let ob =this? this.ob || {}:{};
	ob=Object.defineProperties(ob, Object.getOwnPropertyDescriptors(obj));
	const xx=function xx() {
		return Object.create(ob);
	}
	xx.com=com;
	xx.ob=ob;
        return xx;
}
```
we need avoid use assign, ...spread, for/for in/foreach, almost all of array.prototype, we need change all properties clone code. maybe just like refactor.

**Comment by lornally** [comment]
i have fix it in stampit-org/stamp-specification, but i can not fix it in stampit, because cannot pass the test. this is the code for stampit-org/stamp-specification:
```js
// const assign = Object.assign; fix getter/setter bug.
const assign = (ob, obj) => {
  if (typeof obj !== 'undefined') {
    return Object.defineProperties(ob, Object.getOwnPropertyDescriptors(obj));
  }
  return ob;
};

```

**Comment by koresar** [comment]
Stampit has a set of very robust tests. Let me check why this doesn't work.

**Comment by lornally** [comment]
ok, i will make a PR

**Comment by AHaliq** [comment]
hows progress with the setter? I'm trying to use getters and setters with stampit too.

**Comment by koresar** [comment]
Going well. Thank you for poking. 😀
The coding bit is done. The unit tests left. 
Sorry for the delay. (I'm having a baby girl soon.)

**Comment by lornally** [comment]
congratulations :)

**Comment by koresar** [comment]
Thank you 😀

**Comment by koresar** [comment]
Implemented and published as stampit v4.3

**Comment by koresar** [comment]
@lornally @AHaliq thank you people for poking me from time to time. :) 
The feature is your merit.
